config/connection_settings.go

```diff
@@ -56,8 +56,13 @@ type AzureConnectionSettings struct {
 	*SharedAccessKey
 }
 
-// CreateAzureConnectionSettings creates the configuration data for establishing connection to the Azure IoT Hub.
+// CreateAzureConnectionSettings creates the configuration data for establishing connection to the Azure IoT Hub, using no IDScopeProvider.
```

Comment:
How do you use this method? Do we need it at all?

Reply:
This is the original method. I did not touch its signature as I did not want to go and modify all the tests too, as I wanted the PR to have a little a possible modified lines.
If you are OK with several extra lines with changes in the tests, then this method is not needed.

Reply:
I am not sure that I can understand your concerns here. In the end, we have to improve the code base, the number of modified lines should not be an issue.